Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue2482.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+#if UITEST
+using Xamarin.Forms.Core.UITests;
+using Xamarin.UITest;
+using NUnit.Framework;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(UITestCategories.Animation)]
+#endif
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 2482,
+		"Animating a `View` that is currently animating will throw `System.InvalidOperationException`", 
+		PlatformAffected.All)]
+	public class Issue2482 : TestContentPage
+	{
+		Label _result;
+		int _clicks;
+
+		const string ButtonId = "SpinButton";
+		const string Success = "Success";
+
+		protected override void Init()
+		{
+			var layout = new StackLayout();
+
+			var instructions = new Label { Text = "Tap the button below twice quickly." 
+												+ " If the application crashes, this test has failed." };
+
+			_result = new Label { Text = Success, IsVisible = false };
+
+			var button = new Button
+			{
+				BackgroundColor = Color.Black,
+				TextColor = Color.White,
+				Text = "Quickly Double Tap This Button",
+				HeightRequest = 200,
+				HorizontalOptions = LayoutOptions.Center,
+				VerticalOptions = LayoutOptions.Center,
+				AutomationId = ButtonId
+			};
+
+			button.Clicked += async (sender, args) =>
+			{
+				_clicks += 1;
+
+				await button.RotateTo(539, 3000, Easing.CubicOut);
+				await button.RotateTo(0, 3000, Easing.CubicIn);
```

Comment:
In messing with this I noticed that now it seems to hang on this second await call. If you move the *_clicks += 1;* down below these awaits and then click twice really quickly it never increments the clicks to 2. It should probably still fall through from both clicks yes?